tdrs-frontend/src/components/Header/Header.jsx

```diff
@@ -18,60 +18,59 @@ import NavItem from '../NavItem/NavItem'
  */
 function HeaderComp() {
   const pathname = useSelector((state) => state.router.location.pathname)
-  const [isMenuVisible, setIsMenuVisible] = useState(false)
   const user = useSelector((state) => state.auth.user)
   const authenticated = useSelector((state) => state.auth.authenticated)
 
   const menuRef = useRef()
 
-  let tabIndex = 0
-  /* istanbul ignore next  */
-  const handleTabKey = (e) => {
-    /* istanbul ignore if */
-    if (menuRef.current.classList.contains('is-visible')) {
-      e.preventDefault()
-      console.log('is visible')
-      const focusableMenuElements = [
-        ...menuRef.current.querySelectorAll('button'),
-        ...menuRef.current.querySelectorAll('a'),
-      ]
+  const keyListenersMap = useMemo(() => {
+    let tabIndex = 0
+    /* istanbul ignore next  */
+    const handleTabKey = (e) => {
+      /* istanbul ignore if */
+      if (menuRef.current.classList.contains('is-visible')) {
+        e.preventDefault()
+        const focusableMenuElements = [
+          ...menuRef.current.querySelectorAll('button'),
+          ...menuRef.current.querySelectorAll('a'),
+        ]
 
-      const firstElement = focusableMenuElements[0]
-      const lastIndex = focusableMenuElements.length - 1
-      const lastElement = focusableMenuElements[lastIndex]
+        const lastIndex = focusableMenuElements.length - 1
 
-      if (focusableMenuElements.includes(document.activeElement)) {
-        if (!e.shiftKey && tabIndex >= lastIndex) {
-          tabIndex = 0
-        } else if (e.shiftKey && tabIndex === 0) {
-          tabIndex = lastIndex
-        } else if (e.shiftKey) {
-          tabIndex -= 1
+        if (focusableMenuElements.includes(document.activeElement)) {
+          if (!e.shiftKey && tabIndex >= lastIndex) {
+            tabIndex = 0
+          } else if (e.shiftKey && tabIndex === 0) {
+            tabIndex = lastIndex
+          } else if (e.shiftKey) {
+            tabIndex -= 1
+          } else {
+            tabIndex += 1
+          }
         } else {
-          tabIndex += 1
+          tabIndex = 0
         }
-      } else {
-        tabIndex = 0
+
+        focusableMenuElements[tabIndex].focus()
       }
 
-      focusableMenuElements[tabIndex].focus()
+      return false
     }
 
-    return false
-  }
-
-  const keyListenersMap = new Map([[9, handleTabKey]])
+    return new Map([[9, handleTabKey]])
+  }, [menuRef])
 
   /* istanbul ignore next  */
   useEffect(() => {
     function keyListener(e) {
       const listener = keyListenersMap.get(e.keyCode)
       return listener && listener(e)
     }
+
     document.addEventListener('keydown', keyListener)
 
     return () => document.removeEventListener('keydown', keyListener)
-  }, [])
+  }, [keyListenersMap])
```

Comment:
All of this is just refactored for clarity and to remove unneeded logic and linter warnings